lib/cli/src/Cardano/CLI.hs

```diff
@@ -1721,8 +1845,23 @@ walletStyleOption defaultStyle accepted = option (eitherReader fromTextS)
     prettyStyle s =
         "  " ++ T.unpack (toText s) ++ " (" ++ fmtAllowedWords s ++ ")"
 
+
+instance FromText KeyEncoding where
+    fromText "bech32" = return Bech32Encoding
+    fromText "hex" = return HexEncoding
+    fromText _ = Left $ TextDecodingError "Invalid key encoding option."
```

Comment:
Could have `Codec` directly and skip the intermediary `KeyEncoding´ type, I think